sycl/test/on-device/basic_tests/partition_supported.cpp

```diff
@@ -0,0 +1,92 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+/* Check that:
+1) [info::device::partition_properties]: returns the partition properties
+supported by this SYCL device; a vector of info::partition_property. If this
+SYCL device cannot be partitioned into at least two sub devices then the
+returned vector **must be empty**.
+
+2) [create_sub_devices()]: If the SYCL device
+does not support info::partition_property::partition_by_affinity_domain or the
+SYCL device does not support the info::partition_affinity_domain provided, an
+exception with the **feature_not_supported error code must be thrown**.
+*/
+
+#include <CL/sycl.hpp>
+
+/** returns true if the device supports a particular affinity domain
+ */
+static bool
+supports_affinity_domain(const cl::sycl::device &dev,
+                         cl::sycl::info::partition_property partitionProp,
+                         cl::sycl::info::partition_affinity_domain domain) {
+  if (partitionProp !=
+      cl::sycl::info::partition_property::partition_by_affinity_domain) {
+    return true;
+  }
+  auto supported =
+      dev.get_info<cl::sycl::info::device::partition_affinity_domains>();
+  for (cl::sycl::info::partition_affinity_domain dom : supported) {
+    if (dom == domain) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/** returns true if the device supports a particular partition property
+ */
+static bool
+supports_partition_property(const cl::sycl::device &dev,
+                            cl::sycl::info::partition_property partitionProp) {
+  auto supported = dev.get_info<cl::sycl::info::device::partition_properties>();
+  for (cl::sycl::info::partition_property prop : supported) {
+    if (prop == partitionProp) {
+      return true;
+    }
+  }
+  return false;
+}
+
+int main() {
+
+  auto dev = cl::sycl::device(cl::sycl::default_selector());
+
+  cl::sycl::info::partition_property partitionProperty =
+      cl::sycl::info::partition_property::partition_by_affinity_domain;
+  cl::sycl::info::partition_affinity_domain affinityDomain =
+      cl::sycl::info::partition_affinity_domain::next_partitionable;
+
+  if (supports_partition_property(dev, partitionProperty)) {
+    if (supports_affinity_domain(dev, partitionProperty, affinityDomain)) {
+      auto subDevices = dev.create_sub_devices<
+          cl::sycl::info::partition_property::partition_by_affinity_domain>(
+          affinityDomain);
+
+      if (subDevices.size() < 2) {
+        std::cerr << "device::create_sub_device(info::partition_affinity_"
```

Comment:
I have only one comment regarding the test. Currently in the places where test is expected to fail error message is printed to std::cerr but program still exits successfully with code 0. Can we return non-zero code in such cases to make test fail with the LIT? As far as I understand otherwise test will always pass even though error message is printed (output is not checked with FileCheck),

Reply:
Updated.